projects/plugins/jetpack/extensions/store/membership-products/resolvers.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { store as editorStore } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs, getQueryArg } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import {
+	saveProduct,
+	setApiState,
+	setConnectUrl,
+	setProducts,
+	setShouldUpgrade,
+	setSiteSlug,
+} from './actions';
+import { onError } from './utils';
+import { API_STATE_CONNECTED, API_STATE_NOTCONNECTED } from './constants';
+import getConnectUrl from '../../shared/get-connect-url';
+
+export const getProducts = ( selectedProductId = 0, setSelectedProductId = () => {} ) => async ( {
+	dispatch,
+	registry,
+} ) => {
+	const origin = getQueryArg( window.location.href, 'origin' );
+	const path = addQueryArgs( '/wpcom/v2/memberships/status', {
+		source: origin === 'https://wordpress.com' ? 'gutenberg-wpcom' : 'gutenberg',
+	} );
+
+	try {
+		const response = await apiFetch( { path, method: 'GET' } );
+		if ( ! response && typeof response !== 'object' ) {
+			return;
+		}
+
+		/**
+		 * WP_Error returns a list of errors with custom names:
+		 * `errors: { foo: [ 'message' ], bar: [ 'message' ] }`
+		 * Since we don't know their names, to get the message, we transform the object
+		 * into an array, and just pick the first message of the first error.
+		 *
+		 * @see https://developer.wordpress.org/reference/classes/wp_error/
+		 */
+		const wpError = response?.errors && Object.values( response.errors )?.[ 0 ]?.[ 0 ];
+		if ( wpError ) {
+			dispatch( setApiState( API_STATE_NOTCONNECTED ) );
+			onError( wpError, registry );
+			return;
+		}
+
+		const postId = registry.select( editorStore ).getCurrentPostId();
+
+		dispatch( setConnectUrl( getConnectUrl( postId, response.connect_url ) ) );
+		dispatch( setShouldUpgrade( response.should_upgrade_to_access_memberships ) );
+		dispatch( setSiteSlug( response.site_slug ) );
+
```

Comment:
Here we are missing the upgrade_url value from the response. We use this value to redirect the user to this URL if he doesn't have a valid Jetpack plan. 
I can add this functionality in the Payment button block issue if you prefer.